slyd/slyd/jsonapi/serializers.py

```diff
@@ -0,0 +1,655 @@
+from slyd.jsonapi.utils import (RESOURCE_OBJECT_ORDER, TOP_LEVEL_OBJECT_ORDER,
+                                cached_property, deep_getattr, dasherize,
+                                order_dict, should_include_field,
+                                type_from_model_name)
+from slyd.orm.base import AUTO_PK, Model
+from slyd.orm.exceptions import ImproperlyConfigured
+from slyd.orm.fields import Field as OrmField
+from slyd.orm.relationships import BaseRelationship, HasMany
+
+__all__ = [
+    'JsonApiSerializer',
+    'JsonApiPolymorphicSerializer',
+]
+
+DELETED_PROFILE = 'https://portia.scrapinghub.com/jsonapi/extensions/deleted'
+UPDATES_PROFILE = 'https://portia.scrapinghub.com/jsonapi/extensions/updates'
+DELETED_PROFILE_ALIAS = 'deleted'
+UPDATES_PROFILE_ALIAS = 'updates'
+
+
+class JsonApiSerializerMeta(SchemaMeta):
+    """Meta class for JSON API schemas."""
+    def __new__(mcs, name, bases, attrs):
+        parents = [b for b in bases if isinstance(b, JsonApiSerializerMeta)]
+        if not parents:
+            return super(JsonApiSerializerMeta, mcs).__new__(
+                mcs, name, bases, attrs)
+
+        meta = attrs.pop('Meta', None)
+
+        try:
+            model = meta.model
+        except AttributeError:
+            raise TypeError(
+                u"Class '{}' is missing the 'Meta.model' attribute.".format(
+                    name))
+
+        schema_type = type_from_model_name(model.__name__)
+
+        meta_bases = (meta, object) if meta else (object,)
+        schema_attrs = {
+            'Meta': type('Meta', meta_bases, {
+                'type_': schema_type,
+                'inflect': dasherize,
+            })
+        }
+
+        links = getattr(meta, 'links', {})
+        for attrname, field in iteritems(model._fields):
+            if isinstance(field, OrmField):
+                field_copy = object.__new__(field.__class__)
+                field_copy.__dict__ = dict(field.__dict__)
+                field_copy.load_from = None
+                field_copy.dump_to = None
+                schema_attrs[attrname] = field_copy
+            elif isinstance(field, BaseRelationship):
+                rel_links = links.get(attrname, {})
+                rel_many = isinstance(field, HasMany)
+                rel_options = {
+                    'self_url': rel_links.get('self', ''),
+                    'related_url': rel_links.get('related', ''),
+                    'many': rel_many
+                }
+                if not rel_many:
+                    rel_options['allow_none'] = True
+                if field.polymorphic:
+                    schema_attrs[attrname] = PolymorphicRelationship(
+                        **rel_options)
+                else:
+                    schema_attrs[attrname] = Relationship(
+                        type_=type_from_model_name(field.model.__name__),
+                        id_field='pk',
+                        **rel_options)
```

Comment:
Why do we have to copy over the entire `model._fields` to the serializer class? I think delegating to the model in the serializer is ok as we have a reference to it from the serializer `Meta`. If the issue is with the `links` attribute in the serializer we can make a factory method which will create the appropriate relationships with the relevant relationships options.